Source/ExpressionExtensions.cs

```diff
@@ -202,16 +202,11 @@ public static Expression PartialMatcherAwareEval(this Expression expression)
 
 		private static bool ReturnsMatch(MethodCallExpression expression)
 		{
-			if (!expression.Method.IsDefined(typeof(AdvancedMatcherAttribute), true))
```

Comment:
One would expect there to be a check for the obsolete `[Matcher]` attribute, but such a check wasn't there previously. It might be good to investigate more thoroughly exactly when `ReturnsMatch` gets used.